src/qibo/backends/numpy.py

```diff
@@ -645,9 +645,7 @@ def aggregate_shots(self, shots):
         return self.cast(shots, dtype=shots[0].dtype)
 
     def samples_to_binary(self, samples, nqubits):
-        ### This is faster just staying @ NumPy.
         qrange = np.arange(nqubits - 1, -1, -1, dtype=np.int32)
-        samples = self.to_numpy(samples)
         return np.mod(np.right_shift(samples[:, None], qrange), 2)
```

Comment:
Aren't all the other operations in here still NumPy? Or is Torch automatically converting NumPy calls?

I expected `self.np.` usage, but instead there is `np.` everywhere...

Reply:
I tried `self.np` and it creates other issues because `torch.arange` and `right_shift` work differently

Reply:
But is `np.right_shift()` able to work on Torch tensors? (i.e. taking Torch tensors as input and returning them as output)

I see it could be vaguely possible, but it's still good to know, because I wouldn't have been confident to say it's actually able.

Reply:
```python
In [6]: import numpy as np
   ...: import torch
   ...: 
   ...: x = torch.Tensor(torch.arange(10))
   ...: qrange = np.arange(10 - 1, -1, -1, dtype=np.int32)
   ...: type(np.right_shift(x[:, None], qrange))
Out[6]: torch.Tensor
```

Reply:
Nice, thanks :)

Reply:
It seems that by now this starts being reliable

https://numpy.org/doc/stable/user/basics.interoperability.html#operating-on-foreign-objects-without-converting
https://numpy.org/neps/nep-0018-array-function-protocol.html#nep18